src/wolfgame.py

```diff
@@ -3284,35 +3244,29 @@ def fgoat(cli, nick, chan, rest):
     cli.msg(chan, messages["goat_success"].format(nick, goatact, togoat))
 
 @handle_error
-def return_to_village(cli, chan, nick, show_message):
+def return_to_village(var, chan, target, *, show_message):
     with var.GRAVEYARD_LOCK:
-        if nick in var.DISCONNECTED.keys():
-            hm = var.DISCONNECTED[nick][1]
-            act = var.DISCONNECTED[nick][0]
-            if users.exists(nick):
-                ident = irc_lower(users.get(nick).ident)
-                host = users.get(nick).host.lower()
-                acc = irc_lower(users.get(nick).account)
-            else:
-                acc = None
-            if not acc or acc == "*":
-                acc = None
-            hostmask = ident + "@" + host
-            if (acc and acc == act) or (hostmask == hm and not var.ACCOUNTS_ONLY):
-                del var.DISCONNECTED[nick]
-                var.LAST_SAID_TIME[nick] = datetime.now()
-                for r,rset in var.ORIGINAL_ROLES.items():
-                    if "(dced)"+nick in rset:
-                        rset.remove("(dced)"+nick)
-                        rset.add(nick)
-                if nick in var.DCED_PLAYERS.keys():
-                    var.PLAYERS[nick] = var.DCED_PLAYERS.pop(nick)
+        temp = target.lower()
+        if temp.nick in var.DISCONNECTED:
+            if (temp.account is not None and users.equals(temp.account, account) or
+                temp.userhost is not None and users.equals(temp.userhost, hostmask) and not var.ACCOUNTS_ONLY):
+
+                del var.DISCONNECTED[temp.nick]
+                var.LAST_SAID_TIME[temp.nick] = datetime.now()
+                for roleset in var.ORIGINAL_ROLES.values():
+                    if "(dced)" + temp.nick in roleset:
+                        roleset.remove("(dced)" + temp.nick)
+                        roleset.add(temp.nick)
+
+                if temp.nick in var.DCED_PLAYERS:
+                    var.PLAYERS[temp.nick] = var.DCED_PLAYERS.pop(temp.nick)
+
                 if show_message:
-                    cli.mode(chan, "+v", nick, nick+"!*@*")
-                    cli.msg(chan, messages["player_return"].format(nick))
+                    channels.Main.mode(("+" + hooks.Features["PREFIX"]["+"], target))
```

Comment:
I was thinking the same thing kinda, but we don't do mode changes that much. And you'd need two methods, one for voicing and one for devoicing, or you use a keyword-only argument, and in such cases you can't handle voicing and devoicing in a single change. I prefer that, where it makes sense to, just use .mode instead. If I do add the method though, it won't be as part of the PR (either before or after it's merged).